src/test/suite/ruby.test.ts

```diff
@@ -5,39 +5,157 @@ import * as os from "os";
+
+    assertRubyEnv(rubyEnv);
+
+    if (shouldRemoveDir) {
+      fs.rmSync(dir, { recursive: true, force: true });
     }
 
+    if (originalVersion) {
+      fs.writeFileSync(versionPath, originalVersion);
+    }
+    fs.rmSync(tmpPath, { recursive: true, force: true });
+  });
+
+  test("fetches Ruby environment for .ruby-version using engine", async () => {
+    const tmpPath = fs.mkdtempSync(path.join(os.tmpdir(), "ruby-lsp-test-"));
+    const ruby = new Ruby(
+      {
+        uri: { fsPath: tmpPath },
+      } as vscode.WorkspaceFolder,
+      context,
+    );
+    fs.writeFileSync(path.join(tmpPath, ".ruby-version"), "ruby-3.2.2");
+    const rubyEnv = await ruby.activate();
+
+    assertRubyEnv(rubyEnv);
+    fs.rmSync(tmpPath, { recursive: true, force: true });
+  });
+
+  test("fetches Ruby environment for .ruby-version with .ruby-gemset", async () => {
     const tmpPath = fs.mkdtempSync(path.join(os.tmpdir(), "ruby-lsp-test-"));
+    const ruby = new Ruby(
+      {
+        uri: { fsPath: tmpPath },
+      } as vscode.WorkspaceFolder,
+      context,
+    );
     fs.writeFileSync(path.join(tmpPath, ".ruby-version"), "3.2.2");
+    fs.writeFileSync(path.join(tmpPath, ".ruby-gemset"), "hello");
+    const rubyEnv = await ruby.activate();
 
-    const context = {
-      extensionMode: vscode.ExtensionMode.Test,
-    } as vscode.ExtensionContext;
+    assertRubyEnv(rubyEnv);
+    fs.rmSync(tmpPath, { recursive: true, force: true });
+  });
 
-    ruby = new Ruby(context, {
-      uri: { fsPath: tmpPath },
-    } as vscode.WorkspaceFolder);
-    await ruby.activateRuby(
-      // eslint-disable-next-line no-process-env
-      process.env.CI ? VersionManager.None : VersionManager.Chruby,
+  test("fetches Ruby environment for dev.yml", async () => {
+    const tmpPath = fs.mkdtempSync(path.join(os.tmpdir(), "ruby-lsp-test-"));
+    const ruby = new Ruby(
+      {
+        uri: { fsPath: tmpPath },
+      } as vscode.WorkspaceFolder,
+      context,
     );
+    fs.writeFileSync(path.join(tmpPath, "dev.yml"), "- ruby: '3.2.2'");
+    const rubyEnv = await ruby.activate();
 
-    assert.ok(ruby.rubyVersion, "Expected Ruby version to be set");
-    assert.notStrictEqual(
-      ruby.supportsYjit,
-      undefined,
-      "Expected YJIT support to be set to true or false",
+    assertRubyEnv(rubyEnv);
+    fs.rmSync(tmpPath, { recursive: true, force: true });
+  });
+
+  test("fetches Ruby environment for .tool-versions", async () => {
+    const tmpPath = fs.mkdtempSync(path.join(os.tmpdir(), "ruby-lsp-test-"));
+    const ruby = new Ruby(
+      {
+        uri: { fsPath: tmpPath },
+      } as vscode.WorkspaceFolder,
+      context,
     );
+    fs.writeFileSync(path.join(tmpPath, ".tool-versions"), "ruby 3.2.2");
+    const rubyEnv = await ruby.activate();
+    assertRubyEnv(rubyEnv);
+    fs.rmSync(tmpPath, { recursive: true, force: true });
+  });
 
+  test("fetches Ruby environment for .rtx.toml", async () => {
```

Comment:
Thanks for the heads up. Was the configuration file also renamed to `.mise.toml`?

We'll probably need to accept both for a while since there will be a tail of developers running older versions.

Reply:
> Was the configuration file also renamed to .mise.toml?

Yup it was. (See https://mise.jdx.dev/about.html#about — "Using its .mise.toml config file...")

> We'll probably need to accept both for a while

👍 makes sense